state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -582,6 +609,10 @@ pub mod pallet {
 					T::ChainTracking::get_block_height(),
 				)
 			});
+			used_weight = used_weight.saturating_add(
```

Comment:
Does reading `n` elements really counts as `n` reads? I'm sure, but I would imagine this is more to do with how many lookups by key we do (1 in this case) rather than the size of the data read. (If we do need to count elements, should this be `can_recycle` elements written, or the number of elements left?)

Reply:
This `used_weight` addition is not just for the above mutate, it is for the for loop below, thats why it uses `n` read/writes. Ill add a comment.